internal/btpmanager/credentials/utils_test.go

```diff
@@ -0,0 +1,159 @@
+package btpmgrcreds
+
+import (
+	"fmt"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+type foo struct {
+	x, y int
+}
+
+type boo struct {
+	a, b string
+}
+
+type fakeService struct {
+	reachable bool
+	ticks     int
+}
+
+func (fs *fakeService) fakeCall1() (int, error) {
+	return 91, nil
+}
+
+func (fs *fakeService) fakeCall2() (foo, error) {
+	foo := foo{
+		x: 1,
+		y: 4,
+	}
+	return foo, nil
+}
+
+func (fs *fakeService) fakeCall3() (any, error) {
+	return nil, fmt.Errorf("expected error")
+}
+
+func (fs *fakeService) fakeCall4() (any, error) {
+	return nil, nil
+}
+
+func (fs *fakeService) fakeCallDelayed1() (foo, error) {
```

Comment:
Is there a significant difference between `fakeCallDelayed1` and `fakeCallDelayeds`?
Apart from the the returned structure.